src/mpi4py/futures/_lib.py

```diff
@@ -55,13 +56,42 @@ def setup_mpi_threads():
 
 # ---
 
+class _ExceptionWrapper(Exception):
+    def __reduce__(self):
+        return _unwrap_exc, self.args
 
-def sys_exception():
-    exc = sys.exc_info()[1]
+
+class RemoteTraceback(Exception):
+    pass
+
+
+def _wrap_exc(exc, tb):
+    exc.__cause__ = None
+    exc.__context__ = None
     exc.__traceback__ = None
+    return _ExceptionWrapper(exc, tb)
+
+
+def _unwrap_exc(exc, tb):
+    exc.__cause__ = RemoteTraceback(tb)
     return exc
```

Comment:
nit: could you move this definition to the front, after `RemoteTraceback` and before everything else? I got confused at my first glance "where is this defined"...

Reply:
I'm not sure how exactly you want it defined. I'll do it in the following order

* RemoteTraceback
* _unwrap_exc
* _ExceptionWrapper
* _wrap_exc

so there can be no confusion, things are defined before they are used.

PS: This pattern (use something first in a function/method, define it later in the module) is quite common in Python code.

Reply:
Yeah it's not uncommon, which was why I said it's a nitpick 🙂

Reply:
Thanks.